GunShipAlpha/src/Tests.java

```diff
@@ -35,7 +36,53 @@ void testBehemothLineInteraction() {
         Point end = new Point(450,450);
         Line line = new Line(start,end);
         gameState.addBehemoth(behemoth);
-        boolean isBlocked = behemoth.isIntersectingLine(line);`
+        boolean isBlocked = behemoth.isIntersectingLine(line);
         assertTrue(isBlocked);
     }
+    @Test
+    void testBehemothMovement() {
+        GameState gameState = new GameState();
+        Behemoth behemoth = new Behemoth(350, 350, 50, 50);
+        gameState.addBehemoth(behemoth);
+        double initialDistance = Math.sqrt(Math.pow(350 - 50, 2) + Math.pow(350 - 50, 2));
+        behemoth.moveTowardTarget();
+        Rectangle bounds = behemoth.getBounds();
+        double newDistance = Math.sqrt(Math.pow(bounds.x - 50, 2) + Math.pow(bounds.y - 50, 2));
+        assertTrue(newDistance < initialDistance);
+    }
+    @Test
+    void testRemoveEnemy() {
+        GameState gameState = new GameState();
```

Comment:
the flow here is a bit weird -  create it, remove health, then add it to game state and then remove it? it seems like in reality it would be added before being able to have health removed. thinking about the flow in terms of how things should happen in the actual program is useful for organizing your test flow and making sure it makes sense. if you're testing something weird/unexpected, put a comment in that explains why or mentions it.

Reply:
fixed